### PR TITLE
<main> Convert scale and upgrade test_wl_with_hostPort, nodePort, clusterIp and lb

### PR DESCRIPTION
**Pull request by lscalabrini01**
Backports
v2.9 https://github.com/rancher/rancher/pull/47923
v2.8 https://github.com/rancher/rancher/pull/47924
v2.7 https://github.com/rancher/rancher/pull/47925

### CODE REVIEW

**Review by rancher-max — APPROVED** [review]

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:159]: Thanks!

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:728]: it's good to know that it's possible to disable the servicelb on K3s. I'll add this validation along with the PNI validation ticket to be implemented in the future

follow the ticket 
https://github.com/rancher/qa-tasks/issues/1607

**Review by slickwarren — COMMENTED** [review]
lgtm, waiting on jenkins run for approval.

my one non-blocking nit would be, moving your helpers out to their own file. There are enough helpers to have a dedicated file for

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (anupama2501) [inline tests/v2/validation/networking/connectivity/port_test.go:256]: can we move this to setup?

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:553]: This line is reused in multiple functions, so it would make sense to centralize this logic in a single function

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:559]: This line is reused in multiple functions, so it would make sense to centralize this logic in a single function

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:227]: yes, we can

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:256]: we can't do that. For the upgrade test, we need to check the number of running pods, and to avoid conflicts with other pod tests, I'm using different namespaces for the upgrade tests

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (slickwarren) [inline tests/v2/validation/networking/connectivity/port_test.go:315]: typo here

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (slickwarren) [inline tests/v2/validation/networking/connectivity/port_test.go:726]: nit; the log from this func doesn't seem very helpful. Maybe you could cat the file or something as the log? As-is, it just shows the download status of the curl, which jumbles the logs a bit. (at least I think its from here. Log starting with 
```
Log of the curl command curl ...
```

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (slickwarren) [inline tests/v2/validation/networking/connectivity/port_test.go:566]: nit; still would ask that these helpers get pulled out into a separate file.

**Review by slickwarren — APPROVED** [review]
lgtm, just some nits that could wait if need be.

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (anupama2501) [reply tests/v2/validation/networking/connectivity/port_test.go:557]: I see that this line is reused in multiple functions, but since it’s a simple one liner, keeping it inline in each test might help with readability without adding an extra function. Adding a function could be overkill here, but if you feel strongly about it, we can have it as is. Just a minor suggestion

**Review by floatingman — DISMISSED** [review]
LGTM

File: tests/v2/validation/networking/connectivity/port_test.go
Comment (lscalabrini01) [reply tests/v2/validation/networking/connectivity/port_test.go:568]: This refactor might take some time, so I'll add it to the ticket https://github.com/rancher/qa-tasks/issues/1607 to be addressed in the future, and I'll include this point in the review checklist

**Review by slickwarren — APPROVED** [review]

**Review by anupama2501 — APPROVED** [review]

**Pull request merged**
